File: js/text/fizzbuzz.js
(function ()
{

function range(start, step, end)
{
    var array = [];
    for (start; start < end; start += step)
    {
        array.push(start);
    }
    return array;
}

function print(x)
{
    console.log(x);
}

var isDivisible = function (n)
{
    return function (x)
    {
        return x % n === 0;
    }
}

var isFizz = isDivisible(3);
var isBuzz = isDivisible(5);
var isFizzBuzz = function (x) { return isFizz(x) && isBuzz(x) };

var fizzBuzz = function (n) {
    if (isFizzBuzz(n))
    {
        return "FizzBuzz";
    }
    else if (isFizz(n))
    {
        return "Fizz";
    }
    else if (isBuzz(n))
    {
        return "Buzz";
    }
    else
    {
        return String(n);
    }
}

range(1,1,100).map(function (n) {return fizzBuzz(n)}).forEach(print);

})();

File: hs/text/FizzBuzz.hs
module FizzBuzz where

-- |
isFizz :: Integral a => a -> Bool
isFizz n = n `mod` 3 == 0

-- |
isBuzz :: Integral a => a -> Bool
isBuzz n = n `mod` 5 == 0

-- |
isFizzBuzz :: Integral a => a -> Bool
isFizzBuzz n = isFizz n && isBuzz n

-- |
fizzBuzz :: (Integral a, Show a) => a -> [Char]
fizzBuzz n
    | isFizzBuzz n = "FizzBuzz"
    | isFizz n     = "Fizz"
    | isBuzz n     = "Buzz"
    | otherwise    = show n

-- |
fizzBuzzXs :: (Integral a, Show a) => [a] -> [[Char]]
fizzBuzzXs = map (fizzBuzz)
